lib/conjur/trusted_proxy_filter.rb

```diff
@@ -10,8 +10,8 @@ module Conjur
   #
   # Example: TRUSTED_PROXIES=4.4.4.4,192.168.100.0/24
   class TrustedProxyFilter
-    def initialize(env: ENV, options: { disable_cache: true })
-      @env = env
+    def initialize(config: Conjur::ConjurConfig.new, options: { disable_cache: true })
```

Comment:
I think the name `TrustedProxyFilter` is subtly inaccurate.  A filter is something that takes a list, and returns a sublist (the filtered list) back.  

In contrast, this (via the `call` method) takes a single IP and tells you if it's valid.  It's essentially a big configurable function whose input is an IP and whose output is a boolean value.  What would you name such a function?

Probably something like `valid_ip?`.  However, since this is a class name (which can't end in `?`) I'd consider something like `class IsIpValid` or `class IsIpTrusted`.

In client code:

```ruby
is_trusted = IsIpTrusted.new(...)
if is_trusted(suspicious_ip)
   # etc....
```

Reply:
Submitted a [new issue](https://github.com/cyberark/conjur/issues/2176) for this, don't want to deal w/ it as part of this PR.